### ISSUE TITLE
Newbie issues

### ISSUE THREAD

**Issue by PianSom** [RaymondMouthaan/sonweb-docker#1]
Hi

Apologies for this - my first encounter with docker ...

I have installed docker on a raspi3, lauched your image with 
`docker run -it -v /host/data:/data -p 8080:80 -e APACHE_SERVER_NAME=mysonweb.mydomain --name sonweb raymondmm/sonweb`

and gone to raspi-IP:8080 and am getting the following:
```
Warning: fopen(/app/sonweb/data/devices.csv): failed to open stream: Permission denied in /app/sonweb/includes/top.php on line 48
Can't create file
```

Any suggestions where I may have gone wrong? TIA!

**Comment by RaymondMouthaan** [comment]
Sorry for the late response, didn't see your issue earlier.

Does the /host/data directory exist on your host system?

**Comment by tynor88** [comment]
Same issue as @PianSom

**Comment by tynor88** [comment]
docker log output:

> Clearing any old processes...
>/data/firmwares created.
>/data/updates created.
>Starting apache...

**Comment by RaymondMouthaan** [comment]
@tynor88 have you created a directory on the host ? 

If yes , what’s the command you are using and does it contain the files and directories as mentioned in the log? 

Maybe you can post an ls -al from the host directory? 

On which system are you running? Raspberry Pi?

**Comment by PianSom** [comment]
I abandoned trying to get it to work on my Pi and tried instead on a Mac - where it is working just fine.

I did have the directories (with correct permissions) set up on my Pi.

**Comment by RaymondMouthaan** [comment]
As soon as I can, I look into this further. It should work on rpi because that’s where I use it myself. 

If possible could you try to reproduce the error and post the output of ls -al on the pi directory which should persist the sonweb data files and dirs? 

Thanks in advance

**Comment by RaymondMouthaan** [comment]
Guys, i found the issue! Basically the user apache inside the container hasn't got access to create a file on the data dir because it belongs to root. My mistake and i'll solve it later today.

I'll keep you informed!

**Comment by tynor88** [comment]
Yes exactly. It's a permission issue 👍  Any way to fix it by myself, until you push a new image? @RaymondMouthaan

**Comment by RaymondMouthaan** [comment]
@PianSom  & @tynor88 

I just pushed an new image to [docker hub](https://hub.docker.com/r/raymondmm/sonweb/tags/) which fixes the permission issue. I tested it on my Raspberry PI (arm). Could you confirm if it works for you on rpi and if possible (@PianSom) on mac as well?

**Comment by tynor88** [comment]
@RaymondMouthaan still same issue. Here is the ls from inside the docker container:

> drwxr-xr-x    1 root     root          4096 Mar 27 18:14 .
drwxr-xr-x    1 root     root          4096 Mar 27 18:14 ..
-rwxr-xr-x    1 root     root             0 Mar 27 18:14 .dockerenv
drwxr-xr-x    1 apache   apache        4096 Mar 27 17:21 app
drwxr-xr-x    1 root     root          4096 Mar 27 17:20 bin
drwxr-xr-x    1 root     root          4096 Mar 27 17:21 bootstrap
lrwxrwxrwx    1 apache   apache          16 Mar 27 17:21 data -> /app/sonweb/data
drwxr-xr-x    5 root     root           320 Mar 27 18:15 dev
drwxr-xr-x    1 root     root          4096 Mar 27 18:14 etc
drwxr-xr-x    2 root     root          4096 Nov 30 21:52 home
drwxr-xr-x    1 root     root          4096 Mar 27 17:20 lib
drwxr-xr-x    5 root     root          4096 Nov 30 21:52 media
drwxr-xr-x    2 root     root          4096 Nov 30 21:52 mnt
dr-xr-xr-x  287 root     root             0 Mar 27 18:15 proc
drwx------    1 root     root          4096 Mar 27 18:16 root
drwxr-xr-x    1 root     root          4096 Mar 27 17:21 run
drwxr-xr-x    1 root     root          4096 Mar 27 17:20 sbin
drwxr-xr-x    2 root     root          4096 Nov 30 21:52 srv
dr-xr-xr-x   12 root     root             0 Mar 27 18:15 sys
drwxrwxrwt    1 root     root          4096 Mar 27 18:15 tmp
drwxr-xr-x    1 root     root          4096 Mar 27 17:20 usr
drwxr-xr-x    1 root     root          4096 Mar 27 17:20 var

**Comment by RaymondMouthaan** [comment]
Just to make sure.. Have you pulled the latest image before deploying? 

Using this command :

docker pull raymondmm/sonweb 

This makes sure you don't reuse the previous version of the image

**Comment by PianSom** [comment]
I'm happy to check, but this is the only time I've ever used docker, so can you remind me how to update from github? :)

PS Loving the work, bt the way. I now seem to have acquired 20 (!!) Sonoffs and this utility makes managing them much easier. I suggestion - I'd love to see a column in the detailed view for wificonfig

**Comment by PianSom** [comment]
Ah, beaten to it!!

**Comment by PianSom** [comment]
This version still works on Mac. 

And I see that wificonfig is there - don't know how I missed that!

**Comment by RaymondMouthaan** [comment]
@PianSom have you got it running on both raspberry pi and Mac or just Mac?

If you have suggestions / improvements / bugs regarding sonweb, may I forward you to https://github.com/reloxx13/SonWEB? All credit for @reloxx13 for building a nice piece of software 👍🏼. As for me, I am only providing the docker image for it 😌.

**Comment by PianSom** [comment]
After my starting message above I pretty much gave up on the Pi version, though iirc I didn't delete it so it should still be there. However, I can't seem to get the docker service to start - any suggestions as to what I may have done 3 weeks ago? :) 


```
pi@pi-hole:~ $ sudo systemctl start docker
Failed to start docker.service: Unit docker.service is masked.
pi@pi-hole:~ $ sudo systemctl enable  docker
Synchronizing state of docker.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable docker
Failed to enable unit: Unit file /etc/systemd/system/docker.service is masked.
```

**Comment by PianSom** [comment]
```
pi@pi-hole:~ $ sudo systemctl unmask  docker
Removed /etc/systemd/system/docker.service.
pi@pi-hole:~ $ sudo systemctl enable  docker
docker.service is not a native service, redirecting to systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable docker
pi@pi-hole:~ $ sudo systemctl start docker
Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.
pi@pi-hole:~ $ systemctl status docker.service
● docker.service - LSB: Create lightweight, portable, self-sufficient containers.
   Loaded: loaded (/etc/init.d/docker; generated; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2018-03-27 20:05:53 BST; 14s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 27804 ExecStart=/etc/init.d/docker start (code=exited, status=1/FAILURE)

Mar 27 20:05:53 pi-hole systemd[1]: Starting LSB: Create lightweight, portable, self-sufficient containers....
Mar 27 20:05:53 pi-hole docker[27804]: /usr/bin/dockerd not present or not executable ... failed!
Mar 27 20:05:53 pi-hole systemd[1]: docker.service: Control process exited, code=exited status=1
Mar 27 20:05:53 pi-hole systemd[1]: Failed to start LSB: Create lightweight, portable, self-sufficient containers..
Mar 27 20:05:53 pi-hole systemd[1]: docker.service: Unit entered failed state.
Mar 27 20:05:53 pi-hole systemd[1]: docker.service: Failed with result 'exit-code'.
```

**Comment by tynor88** [comment]
@RaymondMouthaan  yes i did pull the latest. As you can see the data folder inside the container is owned by apache. But it still doesn't work :(

**Comment by tynor88** [comment]
I know why, i fixed it by chowninig the folder, rather than the symlink.

Can you update the dockerfile? 

> chown apache:apache app/sonweb/data

Cheers 👍

**Comment by RaymondMouthaan** [comment]
I am on a 3 hours drive... I will asap, but it doesn't explain why it works on my machine 🤔

**Comment by RaymondMouthaan** [comment]
@tynor88 can you please give it another try? I am not able to test, since i have my docker swarm not available for the comming days ..

@PianSom I am sorry but your issues with docker service is out of scope of this topic. However if you need help to get docker up and running on your rpi, i am willing to help. If possible i recommend saving your data and start from the beginning by formating your sd card and install [rasbian](https://www.raspberrypi.org/downloads/raspbian/) image on it, preferable the lite version if you don't use a user interface ... from there installing docker wouldnt be any rocket science. From where i am now I dont have access to the instruction I wrote but when i get back home I can help you out if you like.

**Comment by PianSom** [comment]
Hi @RaymondMouthaan - actually the only time I use docker is for your code, and it works just fine on my Mac :) I just start up Docker and load the container when I want to do some Sonoff maintenance.

Let me know if you need me to test in that environment.
